### PR TITLE
New IBM MQ Scaler 

### PR DESCRIPTION
**Pull request by jessm12**
As outlined in issue #1253:

Add the Scaler file and switch case for an IBM MQ scaler that scales depending on queue depth of an IBM MQ Queue.
Add the unit tests for this scaler.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Tests have been added
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [x] Changelog has been updated

_(Replaces PR #1254 - fixes DCO/squashed commits)_

### CODE REVIEW

**Comment by zroubalik** [comment]
Could you please update Changelog with this feature? (`New` section)

File: pkg/scalers/ibmmq_scaler.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	}
+	return response.CommandResponse[0].Parameters.Curdepth, nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *IBMMQScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetQueueLengthQty := resource.NewQuantity(int64(s.metadata.targetQueueDepth), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "IBMMQ", s.metadata.queueName)),
```
Comment (zroubalik) [inline pkg/scalers/ibmmq_scaler.go:204]: The metric name should be unique, that means that if you specify two or more IBM MQ scalers in one ScaledObject, each scaler should have a different metric name. I wonder if this is enough?
Is viable a usecase, where we have two MQ scalers with the same queueName but pointing to the different MQ (server)?

File: pkg/scalers/ibmmq_scaler.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
+/**
+* © Copyright IBM Corporation 2020
```
Comment (zroubalik) [inline pkg/scalers/ibmmq_scaler.go:2]: Please remove this copyright

**Comment by cpilton** [comment]
Copyright and Apache 2.0 text comments removed from file headers at the request of the KEDA project so that our contribution presents consistently with the wider codebase.

**Comment by zroubalik** [comment]
Nevermind the DCO for now, I'll try to fix that before the merge

File: pkg/scalers/ibmmq_scaler.go (hunk 1 above)
Comment (jessm12) [reply pkg/scalers/ibmmq_scaler.go:188]: This is a great point @zroubalik and would be a use case for sure, we are adding a solution now where we include the Queue Manager name in the metric name as well as the name of the Queue

**Comment by jeffhollan** [comment]
Discussed in standup, I will help contact CNCF around the license stuff.  We also discussed changed `username` and `password` to `usernameFromEnv` and `passwordFromEnv` and also adding in the code path for those to load up from env / inline secrets

**Comment by caniszczyk** [comment]
FYI for copyright notices in CNCF projects, see https://github.com/cncf/foundation/blob/master/copyright-notices.md

**Comment by cpilton** [comment]
IBM Copyright and Apache 2.0 comments removed from file headers at the request of the KEDA project so that our contribution presents consistently with the wider codebase ‘house style’. In accordance with the KEDA and CNCF contribution guidelines, IBM copyright is unaffected and this contribution is made under Apache 2.0 license.

**Comment by zroubalik** [comment]
@cpilton @jessm12 as I mentioned above, could you please update Changelog with this feature? (`New` section). I think we can proceed with the merge

**Comment by jessm12** [comment]
> @cpilton @jessm12 as I mentioned above, could you please update Changelog with this feature? (`New` section). I think we can proceed with the merge

Getting this sorted now, thanks Zbynek!

**Comment by cpilton** [comment]
We've added a commit to update the changelog as requested 👍  
We have another change to allow username & password from env which we will add in shortly.

**Comment by zroubalik** [comment]
@cpilton okay, let me know once it is done :)

**Comment by zroubalik** [comment]
btw, please follow the naming convetion https://keda.sh/docs/2.0/migration/#improved-flexibility--usability-of-trigger-metadata

**Comment by cpilton** [comment]
@zroubalik, we have added the commit which allows a username & password from env 👍

**Review by zroubalik — APPROVED** [review]

**Pull request merged**